### PR TITLE
GH-36477: [CI][macOS] Ignore brew update failure on crossbow tasks

### LINKED ISSUE #36477: [CI][macOS] brew update fails on some verification tasks
### Describe the bug, including details regarding any error messages, version, and platform.

Similar to: https://github.com/apache/arrow/issues/36402

Some of our verification tasks fail on brew update:
* [verify-rc-source-cpp-macos-amd64](https://github.com/ursacomputing/crossbow/actions/runs/5454951110/jobs/9925738789)
* [verify-rc-source-java-macos-amd64](https://github.com/ursacomputing/crossbow/actions/runs/5454959700/jobs/9925757463)
* [verify-rc-source-python-macos-amd64](https://github.com/ursacomputing/crossbow/actions/runs/5454953383/jobs/9925743043)
* [verify-rc-source-ruby-macos-amd64](https://github.com/ursacomputing/crossbow/actions/runs/5454957775/jobs/9925752796)

### Component(s)

Continuous Integration

### PR DESCRIPTION
**Pull request by raulcd**
### Rationale for this change

Some nightly verification tasks are failing.

### What changes are included in this PR?

Ignore possible errors on `brew update` commands on crossbow tasks.

### Are these changes tested?

Crossbow

### Are there any user-facing changes?

No
* Closes: #36477

### CODE REVIEW

**Comment by raulcd** [comment]
@github-actions crossbow submit *macos*

**Comment by raulcd** [comment]
The brew failures have been fixed. The other CI failures are either expected (binary verification) or unrelated (flaky `arrow-acero-asof-join-node-test`, etc)

File: dev/tasks/macros.jinja [hunk 1]
```diff
@@ -204,7 +204,7 @@ on:
     env:
       ARROW_FORMULA: ./arrow/dev/tasks/homebrew-formulae/{{ formula }}
     run: |
-      brew update
+      brew update || :
```
Comment (assignUser) [inline dev/tasks/macros.jinja:207]: Do we maybe want to echo that there was an issue with brew update?

Comment (raulcd) [reply dev/tasks/macros.jinja:207]: you mean something like: `brew update || echo "brew update did not finish successfully"`?

**Review by kou — APPROVED** [review]
+1

**Pull request merged**